geth/api/api.go

```diff
@@ -52,65 +53,88 @@ func (api *StatusAPI) TxQueueManager() *transactions.Manager {
 
 // StartNode start Status node, fails if node is already started
 func (api *StatusAPI) StartNode(config *params.NodeConfig) error {
-	nodeStarted, err := api.b.StartNode(config)
-	if err != nil {
-		return err
-	}
-	<-nodeStarted
-	return nil
+	return api.b.StartNode(config)
 }
 
 // StartNodeAsync start Status node, fails if node is already started
 // Returns immediately w/o waiting for node to start (see node.ready)
 func (api *StatusAPI) StartNodeAsync(config *params.NodeConfig) (<-chan struct{}, error) {
-	return api.b.StartNode(config)
+	resp := make(chan struct{})
+	if api.b.IsNodeRunning() {
+		close(resp)
+		return resp, nil
```

Comment:
It looks like this will make `StartNode` and `StartNodeAsync` return different results when trying to start an already running node, right? In the first case, it will return an error and in the latter it won't.